src/backend/metal/src/command.rs

```diff
@@ -453,176 +147,239 @@ impl command::Buffer<Resources> for CommandBuffer {
     }
 
     fn bind_samplers(&mut self, ss: &[pso::SamplerParam<Resources>]) {
+        use std::f32;
+
         for &stage in [Stage::Vertex, Stage::Pixel].iter() {
-            let mut samplers = [native::Sampler(MTLSamplerState::nil()); MAX_SAMPLERS];
             let mask = stage.into();
-            let mut count = 0;
-            for sm in ss.iter() {
-                if sm.1.contains(mask) {
-                    samplers[sm.2 as usize] = sm.0;
-                    count += 1;
+            for sampler in ss.iter() {
+                if sampler.1.contains(mask) {
+                    // TODO(fkaa): what about min/max lods?
```

Comment:
We definitely [pass those](https://github.com/gfx-rs/gfx/blob/a015a3a6e0dac56d877e54a5427d28714ca30e72/src/core/src/texture.rs#L424) in `SamplerInfo`, so we'll need to use them here. TODO